lambda/aws-rag-appsync-stepfn-opensearch/s3_file_transformer/src/helpers/utils.py

```diff
@@ -0,0 +1,144 @@
+#
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+#
+
+import os
+import boto3
+import json
+from botocore.exceptions import ClientError
+from aws_lambda_powertools.metrics import MetricUnit
+
+from helpers.pdf_transformer import pdf_transformer
+from helpers.image_transformer import image_transformer
+from langchain_core.prompts import PromptTemplate
+from aws_lambda_powertools import Logger, Tracer, Metrics
+
+
+
+
+rekognition=boto3.client('rekognition')
+s3 = boto3.resource('s3')
+
+logger = Logger(service="INGESTION_FILE_TRANSFORMER")
+tracer = Tracer(service="INGESTION_FILE_TRANSFORMER")
+metrics = Metrics(namespace="ingestion_pipeline", service="INGESTION_FILE_TRANSFORMER")
+
+@tracer.capture_method
+def isvalid_file_format(file_name: str) -> bool:
+    file_format = ['.pdf','.txt','.jpg','.png','.jpeg','.svg']
+    if file_name.endswith(tuple(file_format)):
+        return True
+    else:
+        print(f'Invalid file format :: {file_format}')
+        return False
+    
+@tracer.capture_method
+def transform_pdf_document(input_bucket: str,file_name: str,output_bucket: str,output_file_name:str):
+        document_content = pdf_transformer(input_bucket,file_name).load()
+        if not document_content:
+                return 'Unable to load document'             
+        else:
+                encoded_string = document_content.encode("utf-8")
+                s3.Bucket(output_bucket).put_object(Key=output_file_name, Body=encoded_string) 
```

Comment:
what if this fails ?

Reply:
done 6cbb5e0c8ba77186876a170ee7fc58a5f4335ccb